libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -49,10 +49,37 @@ do {                                            \
                        OUTOFGEOFENCE)
 
 
+static uint8_t counter = 0;     // For Debuging
+
+
 AP_GPS_SBF::AP_GPS_SBF(AP_GPS &_gps, AP_GPS::GPS_State &_state,
-                       AP_HAL::UARTDriver *_port) :
+                       AP_HAL::UARTDriver *_port, uint8_t _asterx_setup) :
     AP_GPS_Backend(_gps, _state, _port)
 {
+    switch(_asterx_setup) {
```

Comment:
You don't actually need any of this. You can fingerprint the receiver by asking it for it's capabilities (`grc`) then if it has `INS` it's the -i variant, if it has `Heading` then it's the dual antenna.

Reply:
Yes that is indeed the thing. It is possible that grc gives back that is has the heading and INS. Would it not be more complex? Like if the receiver has both capabilities we have to prefer the heading of the INSNavGeod message and not the one coming from the AttEueler message.

Reply:
That's a good question. I actually would have assumed the opposite way if it had both, so I guess we can keep the different argument types. It is a pattern we've used more recently so it's definitely not unprecedented.